### PR TITLE
MINIFI-313: Makes provenance repository configurable.

### PR DESCRIPTION
**Pull request by jzonthemtn**
Thank you for submitting a contribution to Apache NiFi - MiNiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi-minifi folder?
- [X] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under minifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under minifi-assembly?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ProvenanceRepositorySchema.java [hunk 1]
```diff
@@ -25,27 +25,38 @@
 import static org.apache.nifi.minifi.commons.schema.common.CommonPropertyKeys.PROVENANCE_REPO_KEY;
 
 public class ProvenanceRepositorySchema extends BaseSchema implements WritableSchema {
-    public static final String PROVENANCE_REPO_ROLLOVER_TIME_KEY = "provenance rollover time";
 
+    public static final String PROVENANCE_REPO_ROLLOVER_TIME_KEY = "provenance rollover time";
     public static final String DEFAULT_PROVENANCE_ROLLOVER_TIME = "1 min";
 
+    public static final String PROVENANCE_REPOSITORY_KEY = "repository";
```
Comment (apiri) [inline minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ProvenanceRepositorySchema.java]: maybe implementation would be a better term as it is the provenance repository regardless?

File: minifi-bootstrap/src/test/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformerTest.java [hunk 2]
```diff
@@ -352,6 +354,26 @@ public void doesTransformOnMinimal() throws Exception {
...
     @Test
+    public void doesTransformOnProvenanceRepository() throws Exception {
+        ConfigTransformer.transformConfigFile("./src/test/resources/config-provenance-repository.yml", "./target/");
+        File nifiPropertiesFile = new File("./target/nifi.properties");
+
+        assertTrue(nifiPropertiesFile.exists());
+        assertTrue(nifiPropertiesFile.canRead());
+
+        String nifi = FileUtils.readFileToString(nifiPropertiesFile, Charset.defaultCharset());
+        assertTrue(nifi.contains("nifi.provenance.repository.implementation=org.apache.nifi.provenance.MiNiFiPersistentProvenanceRepository"));
```
[the remark this answers is not in the document]
Comment (jzonthemtn) [reply minifi-bootstrap/src/test/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformerTest.java]: Good point.

File: minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ProvenanceRepositorySchema.java (hunk 1 above)
Comment (jzonthemtn) [reply minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/ProvenanceRepositorySchema.java]: I initially had it as `repository.implementation` but decided that might be unnecessarily long. Simply `implementation` does seem a better fit.

**Comment by apiri** [comment]
reviewing updates

**Comment by apiri** [comment]
changes look good, will get this merged and then evaluate the no op PR as well

**Pull request closed without merging**